barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/ipa.hpp

```diff
@@ -472,11 +493,15 @@ template <typename Curve_> class IPA {
 
         Fr b_zero = Fr(1);
         Fr challenge = opening_claim.opening_pair.challenge;
-        for (size_t i = 0; i < log_poly_length; i++) {
-            b_zero *= Fr(1) + (round_challenges_inv[log_poly_length - 1 - i] * challenge);
-            if (i != log_poly_length - 1)
+        for (size_t i = 0; i < CONST_ECCVM_LOG_N; i++) {
+            stdlib::bool_t<typename Curve::Builder> dummy_round = stdlib::witness_t(builder, i < CONST_ECCVM_LOG_N - log_poly_length);
+
+            Fr monomial = Fr::conditional_assign(dummy_round, Fr(0), round_challenges_inv[CONST_ECCVM_LOG_N - 1 - i] * challenge);
+            b_zero *= Fr(1) + monomial;
+            info("rec verify b_zero after i: ", i, " is ", b_zero.get_value());
+            if (i != CONST_ECCVM_LOG_N - 1) // this if is fine because the number of iterations is constant
```

Comment:
typo in comment?

Reply:
I meant this if statement is fine (as opposed to using a constexpr if or anything)